apps/ledger-live-desktop/src/renderer/modals/EditTransaction/Body.jsx

```diff
@@ -0,0 +1,271 @@
+      updateAccountWithUpdater(mainAccount.id, account =>
+        addPendingOperation(account, optimisticOperation),
+      );
+      setOptimisticOperation(optimisticOperation);
+      setTransactionError(null);
+    },
+    [account, parentAccount, updateAccountWithUpdater],
+  );
+
+  const handleStepChange = useCallback(e => onChangeStepId(e.id), [onChangeStepId]);
+  const error = transactionError || bridgeError;
+  const mainAccount = getMainAccount(account, parentAccount);
+  const feePerGas = new BigNumber(
+    EIP1559ShouldBeUsed(mainAccount.currency)
+      ? params.transactionRaw.maxFeePerGas
+      : params.transactionRaw.gasPrice,
+  );
+  const feeValue = new BigNumber(
+    params.transactionRaw.userGasLimit || params.transactionRaw.estimatedGasLimit,
+  )
+    .times(feePerGas)
+    .div(new BigNumber(10).pow(mainAccount.unit.magnitude));
+  const haveFundToCancel = mainAccount.balance.gt(feeValue.times(1.3));
+  const haveFundToSpeedup = mainAccount.balance.gt(
+    feeValue.times(1.1).plus(account.type === "Account" ? params.transactionRaw.amount : 0),
+  );
+  let isOldestEditableOperation = true;
+  account.pendingOperations.forEach(operation => {
+    if (isEditableOperation(account, operation)) {
+      if (operation.transactionSequenceNumber < params.transactionSequenceNumber) {
+        isOldestEditableOperation = false;
+      }
+    }
+  });
+  const [editType, setEditType] = useState(
+    isOldestEditableOperation && haveFundToSpeedup ? "speedup" : haveFundToCancel ? "cancel" : "",
+  );
+  const handleSetEditType = useCallback(editType => setEditType(editType), []);
+
+  const stepperProps = {
+    title: t("operation.edit.title"),
+    stepId,
+    steps,
+    device,
+    account,
+    parentAccount,
+    transaction,
+    signed,
+    currencyName,
+    hideBreadcrumb: false,
+    error,
+    status,
+    bridgePending,
+    optimisticOperation,
+    openModal,
+    onClose,
+    setSigned,
+    closeModal: handleCloseModal,
+    onChangeTransaction: setTransaction,
+    onRetry: handleRetry,
+    onStepChange: handleStepChange,
+    onOperationBroadcasted: handleOperationBroadcasted,
+    onTransactionError: handleTransactionError,
+    updateTransaction,
+    onConfirmationHandler: params.onConfirmationHandler,
+    onFailHandler: params.onFailHandler,
+    setEditType: handleSetEditType,
+    editType,
+    setIsNFTSend,
+    isNFTSend,
+    transactionRaw: params.transactionRaw,
+    isNftOperation: params.isNftOperation,
+    transactionSequenceNumber: params.transactionSequenceNumber,
+    haveFundToSpeedup,
+    haveFundToCancel,
+    isOldestEditableOperation,
+  };
+
+  return (
+    <Stepper {...stepperProps}>
+      {stepId === "confirmation" ? null : <SyncSkipUnderPriority priority={100} />}
+    </Stepper>
+  );
+};
+
+const m: React$ComponentType<OwnProps> = compose(
+  connect(mapStateToProps, mapDispatchToProps),
+  withTranslation(),
+)(Body);
```

Comment:
I don't know if it works with compose (I guess it does), but can you try a `memo(Body)` here instead of `Body` ?

Reply:
fixed